### PR TITLE
[Core] Optimize Qwen2.5-Omni Token2Wav buffer loading

### LINKED ISSUE #7060: Qwen2.5-Omni Token2Wav rebuilds state_dict once per buffer
## Environment

| Item | Value |
| --- | --- |
| Model | `Qwen/Qwen2.5-Omni-3B` |
| GPU | 1 AMD Instinct MI300X VF |
| PyTorch | `2.12.0+git6bbd260` |
| ROCm | `7.2.53211` |
| vLLM | `0.28.0+rocm723` |
| vLLM-Omni | `0.28.1.dev47` |

All three stages used device 0 with GPU memory utilization values of 0.35, 0.25, and 0.15. The model files were already cached.

## Issue

`Qwen2_5OmniToken2WavForConditionalGenerationVLLM.find_all_registers()` calls `state_dict()` inside the loop over `named_buffers()`:

```python
for name, buf in self.named_buffers():
    if name in self.state_dict():
        registers[name] = {"name": name, "buffer": buf}
```

The membership check has a valid purpose. `named_buffers()` includes persistent and nonpersistent buffers, while `state_dict()` includes only persistent buffers. Calling `state_dict()` for every buffer rebuilds the full state dictionary and repeats the same module traversal.

Qwen2.5-Omni made 219 `state_dict()` calls during each measured startup.

## Proposed change

Collect the fully qualified names of nonpersistent buffers once, then walk `named_buffers()` once:

```python
non_persistent_names = set()
for module_name, module in self.named_modules():
    for buffer_name in getattr(module, "_non_persistent_buffers_set", set()):
        name = f"{module_name}.{buffer_name}" if module_name else buffer_name
        non_persistent_names.add(name)

for name, buffer in self.named_buffers():
    if name not in non_persistent_names:
        registers[name] = {"name": name, "buffer": buffer}
```

The change preserves the original persistent buffer behavior and removes all 219 calls to `state_dict()`.

## Measurements

The benchmark ran one warmup per version and then five measured runs per version. Runs alternated between the baseline and optimized commits. Warmups were excluded from the medians.

| Measurement | Baseline median | Optimized median | Change |
| --- | ---: | ---: | ---: |
| Buffer discovery | 456.946 ms | 1.564 ms | 455.382 ms faster, 99.66% |
| Token2Wav weight loading | 1099.526 ms | 620.436 ms | 479.090 ms faster, 43.57% |
| Startup to `/health` | 105.738 s | 104.322 s | 1.416 s faster, 1.34% |

The startup ranges overlapped, from 104.214 to 106.067 seconds for the baseline and from 103.660 to 105.981 seconds after the change. The direct Token2Wav measurements show the improvement. The full startup measurement includes about 104 seconds of other work and does not isolate the change from normal run variation.

## Why this change

vLLM already checks `_non_persistent_buffers_set` when it collects persistent registered buffers. vLLM-Omni uses the same check in diffusion weight loaders. Applying the existing pattern here avoids a new abstraction and keeps the same buffer selection behavior.

### PR DESCRIPTION
**Pull request by akshatvishu**
Resolves #7060.

#### Summary
`find_all_registers()` called `state_dict()` once for every named buffer. Qwen2.5-Omni made 219 calls during each startup, even though every call traversed the same module tree.

The loader now collects nonpersistent buffer names from `_non_persistent_buffers_set` and walks the named buffers once. The test checks that persistent child buffers remain included, nonpersistent buffers remain excluded, and `state_dict()` is not called.

The code follows the existing vLLM buffer loading pattern. vLLM-Omni also uses the same check in its diffusion weight loaders.

#### Benchmark

| Item | Value |
| --- | --- |
| Model | `Qwen/Qwen2.5-Omni-3B` |
| GPU | 1 AMD Instinct MI300X VF |
| GPU memory | 192 GB |
| PyTorch | `2.12.0+git6bbd260` |
| ROCm | `7.2.53211` |
| vLLM | `0.28.0+rocm723` |
| vLLM-Omni | `0.28.1.dev47` |

The server was started with:

```bash
MIOPEN_FIND_MODE=FAST vllm serve Qwen/Qwen2.5-Omni-3B --omni --host 127.0.0.1 --port 8091 --deploy-config /tmp/qwen2_5_omni_rocm_single_gpu.yaml
```

All three stages used device 0 with GPU memory utilization values of 0.35, 0.25, and 0.15. The benchmark used one warmup per version and five measured runs per version. The measured runs alternated between the baseline and optimized commits, and the model files were already cached.

| Measurement | Baseline median | Optimized median | Change |
| --- | ---: | ---: | ---: |
| Calls to `state_dict()` | 219 | 0 | Removed 219 calls |
| Buffer discovery | 456.946 ms | 1.564 ms | 455.382 ms faster, 99.66% |
| Token2Wav weight loading | 1099.526 ms | 620.436 ms | 479.090 ms faster, 43.57% |
| Startup to `/health` | 105.738 s | 104.322 s | 1.416 s faster, 1.34% |

The startup ranges overlapped, so the PR claims the direct reduction in Token2Wav loading work rather than a stable full startup improvement.

The logged GPU memory figures for stages 0 and 1 were unchanged. Stage 0 used 9.94 GiB for weights and non-PyTorch memory with 2.49 GiB of peak activation memory. Stage 1 used 1.95 GiB with 1.18 GiB of peak activation memory. No GPU memory improvement is claimed.

#### Test

```text
2 passed, 14 warnings in 0.13s
```

Command:

```bash
.venv/bin/pytest tests/model_executor/models/qwen2_5_omni/test_weight_loading.py --run-level advanced_model -q
```

`pre-commit run --files` passed for both changed files.

### CODE REVIEW

**Comment by akshatvishu** [comment]
Self-review: The change is limited to persistent buffer discovery in Token2Wav and replaces 219 repeated `state_dict()` traversals with the `_non_persistent_buffers_set` pattern already used by vLLM. The regression test covers root and nested persistent buffers, excludes a nonpersistent buffer, and verifies that `state_dict()` is never called; both targeted tests and all applicable local pre-commit hooks passed. Five alternating ROCm runs, after one warmup per version, reduced median buffer discovery from 456.946 ms to 1.564 ms and Token2Wav weight loading from 1099.526 ms to 620.436 ms, while the server loaded the same 721 weights and found the same persistent buffer. The full startup ranges overlap, so the PR does not claim a stable end-to-end startup improvement; output quality was not compared because the change only alters how the same persistent buffer names are selected during loading.

**Comment by akshatvishu** [comment]
@vllm-omni-review-bot  All the claims, reason for bottleneck as well as A/B testing is already provided in the pr body itself

**Review by linyueqian — APPROVED** [review]
Correct and cheap. `state_dict()` excludes exactly the buffers whose names sit in their module's `_non_persistent_buffers_set`, so walking `named_modules()` once to collect those qualified names and filtering `named_buffers()` against the set reproduces the old membership test without rebuilding the state dict per buffer; it is the same pattern vLLM's own loaders and this repo's diffusion weight loaders already use. The test pins persistent root and nested buffers included, a non-persistent one excluded, and zero `state_dict()` calls. The A/B numbers in the description are from ROCm on an older vLLM, which is fine for a startup-time change of this shape.

Static read at `40ccac04`; fork head, no PR code executed. pre-commit and DCO are green; the PR has no `ready` label so no general lane has run, and it needs one before merge.
